.github/workflows/black.yml

```diff
@@ -1,11 +1,31 @@
 name: Black Formatter
-on: [push, pull_request]
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches_ignore: []
+
 jobs:
-  black:
+  formatting:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v1
-      - name: Black Code Formatter
-        uses: lgeiger/black-action@master
+
+      - name: Setup black linter
+        run: conda create --quiet --name black black
+
+      - name: Lint python code
+        run: |
+          export PATH="/usr/share/miniconda/bin:$PATH"
+          source activate black
+          black --check buildtest tests/*.py
```

Comment:
i think ``black --check .`` should be sufficient it will autoformat all python files recursively. So no need to specify the directory paths unless we only want to black specific files. You can test this locally and you will get idea.

Reply:
I think it's good to be explicit - it's a relatively simple structure now, so it makes sense to not opt in everything when we don't know what else could be there. We can change this if there is strong rationale to do so.